localstack/config.py

```diff
@@ -905,12 +973,20 @@ def __iter__(self):
 SERVICE_PROVIDER_CONFIG.load_from_environment()
 
 # initialize directories
-if is_in_docker:
-    dirs = Directories.for_container()
-else:
-    dirs = Directories.from_config()
+if LEGACY_DIRECTORIES:
+    if is_in_docker:
+        dirs = Directories.legacy_for_container()
+    else:
+        dirs = Directories.legacy_from_config()
 
-dirs.mkdirs()
+    dirs.mkdirs()
+
+else:
+    if is_in_docker:
+        dirs = Directories.for_container()
+        dirs.mkdirs()
+    else:
+        dirs = Directories.for_host()
```

Comment:
Should the `mkdirs` maybe be executed for the host paths as well?

Reply:
yep great point. they shouldn't in fact, because `config.py` is also executed when running the CLI, and `Directories` should only be associated with the localstack runtime, not simply importing the config.

the comment made me think and i actually removed the `dirs.mkdirs()` call from line 987 as well, and rather initialize the filesystem hierarchy when starting the infra (and create the base directories in the Dockerfile). let's see if that works.